browser/ai_chat/enabled_state_transition_service.h

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) 2025 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#ifndef BRAVE_BROWSER_AI_CHAT_ENABLED_STATE_TRANSITION_SERVICE_H_
+#define BRAVE_BROWSER_AI_CHAT_ENABLED_STATE_TRANSITION_SERVICE_H_
+
+#include "base/memory/raw_ptr.h"
+#include "build/build_config.h"
+#include "components/keyed_service/core/keyed_service.h"
+#include "components/prefs/pref_change_registrar.h"
+
+class PrefService;
+class Profile;
+
+namespace content {
+class BrowserContext;
+}
+
+#if !BUILDFLAG(IS_ANDROID)
+namespace sidebar {
+class SidebarService;
+}
+#endif
+
+namespace ai_chat {
+
+class EnabledStateTransitionService : public KeyedService {
```

Comment:
I would not create a separate KeyedService for this because you need to keep it in sync with AIChatService in terms of which profiles it is enabled for, but also you should be using ProfileKeyedService for things in brave/browser

Reply:
if you use dependency injection you can move this class to components/ai_chat and pass it into AIChatService in the factory